components/Input/index.tsx

```diff
@@ -1,77 +1,92 @@
+import { useField } from 'formik'
+
 import {
 	Grid,
 	IconButton,
 	InputAdornment,
-	SxProps,
-	TextField,
-	Theme,
+	FormControl,
+	InputLabel,
+	OutlinedInput,
+	FormHelperText,
 } from '@mui/material'
 import { Visibility, VisibilityOff } from '@mui/icons-material'
+import CheckIcon from '@mui/icons-material/Check'
+import ClearIcon from '@mui/icons-material/Clear'
+import { useTheme } from '@mui/material/styles'
 
 export interface InputProps {
 	name: string
 	label: string
-	value?: string
-	type?: string
-	sx?: SxProps<Theme> | undefined
-	disabled?: boolean
-	handleShowPassword?: () => void
-	half?: boolean
-	autoFocus?: boolean
-	onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
+	type: 'text' | 'password'
+	half: boolean
 }
 
-const Input = ({
-	name,
-	label,
-	half,
-	autoFocus,
-	type,
-	handleShowPassword,
-	value,
-	sx,
-	disabled,
-	onChange,
-}: InputProps) => (
-	<Grid item xs={12} sm={half ? 6 : 12}>
-		<TextField
-			onChange={onChange}
-			name={name}
-			value={value}
-			sx={sx}
-			disabled={disabled}
-			variant="outlined"
-			fullWidth
-			label={label}
-			autoFocus={autoFocus}
-			type={type}
-			InputProps={
-				name === 'password'
-					? {
-							endAdornment: (
-								<InputAdornment position="end">
-									<IconButton onClick={handleShowPassword}>
-										{type === 'password' ? <Visibility /> : <VisibilityOff />}
-									</IconButton>
-								</InputAdornment>
-							),
-					  }
-					: undefined
-			}
-		/>
-	</Grid>
-)
+const Input: React.FC<InputProps> = ({ ...props }) => {
+	const [field, meta] = useField(props)
+	const theme = useTheme()
+
+	const isError = Boolean(meta.error && meta.touched)
+	const isSuccess = Boolean(!meta.error && field.value)
+
+	const [toggleShowPassword, setToggleShowPassword] = useState<boolean>(false)
+
+	const handleToggleShowPassword = () => {
+		setToggleShowPassword((togglePassword) => !togglePassword)
+	}
+
+	const PasswordIcon = props.type === 'password' && (
+		<IconButton
+			aria-label="toggle password visibility"
+			onClick={handleToggleShowPassword}
+			edge="end"
+		>
+			{toggleShowPassword ? <VisibilityOff /> : <Visibility />}
+		</IconButton>
+	)
+
+	const TextIcon = (isSuccess || isError) && props.type !== 'password' && (
+		<IconButton aria-label="verify check" edge="end">
+			{isError ? (
+				<ClearIcon sx={{ color: theme.custom.offlineRed }} />
+			) : (
+				<CheckIcon sx={{ color: theme.custom.green }} />
+			)}
+		</IconButton>
+	)
+
+	const inputType =
+		// eslint-disable-next-line no-nested-ternary
+		props.type !== 'password'
+			? props.type
+			: toggleShowPassword
+			? 'text'
+			: 'password'
 
-Input.defaultProps = {
-	sx: {},
-	disabled: false,
-	handleShowPassword: () => {},
-	half: false,
-	autoFocus: false,
-	type: 'text',
-	value: '',
-	onChange: () => {},
+	return (
+		<Grid item xs={12} sm={props.half ? 6 : 12}>
+			<FormControl sx={{ width: '100%' }} variant="outlined" error={isError}>
```

Comment:
Is it better if you use TextField instead? It contains all the components/properties you will need that is enwrapped with form control?

Reply:
Yeah, I get you but I did a little bit of customization for input adornment and error messages, and from the docs here: https://mui.com/material-ui/react-text-field/#input-adornments, I wanted to add so more customization and from the docs, I followed through.

Reply:
Ok I'll merge it. Ill give you a heads up on a update on my branch 👍

Reply:
What do you mean by reuse?